zk/lock.go

```diff
@@ -47,13 +53,23 @@ func (l *Lock) Lock() error {
 		return ErrDeadlock
 	}
 
+	if seqZnodePath, ok := mapEphermeralSequenceLockPath[l.path]; ok && seqZnodePath != "" {
+		// Check whether lock has been acquired previously and it still exists
+		if(lockExists(l.c,l.path,seqZnodePath)) {
+			return nil
+		}
+	}
+
 	prefix := fmt.Sprintf("%s/lock-", l.path)
 
 	path := ""
 	var err error
-	for i := 0; i < 3; i++ {
+	tryLock: for i := 0; i < 3; i++ {
```

Comment:
format this file pls :) tryLock should be on a separate line

Reply:
done. Please check